app/javascript/frontend/scripts/view_statistics_chart.js

```diff
@@ -38,10 +38,33 @@ function drawChart (selection, data) {
     .domain(d3.extent(data, d => d[2]))
     .range([height - margin.bottom, margin.top])
 
+  const formatMillisecond = d3.timeFormat('.%L')
+  const formatSecond = d3.timeFormat(':%S')
+  const formatMinute = d3.timeFormat('%I:%M')
+  const formatHour = d3.timeFormat('%I %p')
+  const formatDay = d3.timeFormat('%a %d')
+  const formatWeek = d3.timeFormat('%b %d')
+  const formatMonth = d3.timeFormat('%b')
+  const formatYear = d3.timeFormat('%Y')
+
+  // Custom multi-format override for the x-axis (time/date) tick labels.
+  // This is used to fix some rendering issues encountered when using the default tick formatter.
+  // Based on example from https://github.com/d3/d3-time-format#d3-time-format
+  const multiFormat = function (date) {
+    return (d3.timeSecond(date) < date ? formatMillisecond
```

Comment:
@awead - I am guessing Code Climate doesn't like this function very much. 😁 It is kinda brutal to read.

Should I convert it to a conventional `if/else`?

Reply:
@nicktiberi Actually, the original example isn't _too_ bad. It's basically just a big switch operation. If it were me, I'd probably pull the whole `multiFormat` thing out into its own function so that it was set apart (visually) from the rest of the `drawChart` function. See `findNearestDataPoint` for an comparison example. If it isn't clear from reading the funtion, I'll put in some YARD docs to explain what's coming in and out:

``` js
// @param [Date]
// @return [String] ??
// @notes Custom multi-format override for the x-axis (time/date) tick labels.
function multiformat (date) {
...
```

When this was written, we didn't have a testing framework setup, but we do now. Would you consider adding a jest test for this one function? Don't worry about this rest of the code... It's also possible that getting it strapped into a jest setup may be more trouble that it's worth.

Reply:
If CodeClimate still barks at us, we can just tell it to sit down.

Reply:
I think CodeClimate is still mad about the nested ternary operators, but it didn't like my refactored `if/else` either (too many lines in a function + too many return statements). So maybe nested ternary is OK?

Wrote a few tests for this which seem to cover the cases that we would encounter with the Scholarsphere analytics data, but please let me know if there are some more scenarios I should account for.